Tests/OpenCombineTests/PublisherTests/MapErrorTests.swift

```diff
@@ -0,0 +1,251 @@
+//
+//  MapErrorTests.swift
+//  
+//
+//  Created by Joseph Spadafora on 7/4/19.
+//
+
+import XCTest
+
+#if OPENCOMBINE_COMPATIBILITY_TEST
+import Combine
+#else
+import OpenCombine
+#endif
+
+@available(macOS 10.15, *)
+final class MapErrorTests: XCTestCase {
+    static let allTests = [
+        ("testEmpty", testEmpty),
+        ("testError", testError),
+        ("testRange", testRange),
+        ("testNoDemand", testNoDemand),
+        ("testDemandSubscribe", testDemandSubscribe),
+        ("testDemandSend", testDemandSend),
+        ("testCompletion", testCompletion),
+        ("testCancel", testCancel),
+        ("testCancelAlreadyCancelled", testCancelAlreadyCancelled),
+        ("testLifecycle", testLifecycle),
+    ]
+
+    func testEmpty() {
+        // Given
+        let tracking = TrackingSubscriberBase<Int, OtherError>(
+            receiveSubscription: { $0.request(.unlimited) }
+        )
+        let publisher = PassthroughSubject<Int, TestingError>()
+        // When
+        publisher.mapError { OtherError($0) }.subscribe(tracking)
+        // Then
+        XCTAssertEqual(tracking.history, [.subscription(Subscriptions.empty)])
+    }
+
+    func testError() {
+        // Given
+        let expectedError = TestingError.oops
+        let tracking = TrackingSubscriberBase<Int, OtherError>(
+            receiveSubscription: { $0.request(.unlimited) }
+        )
+        let publisher = PassthroughSubject<Int, TestingError>()
+        // When
+        publisher.mapError { OtherError($0) }.subscribe(tracking)
+        publisher.send(completion: .failure(expectedError))
+        // Then
+        XCTAssertEqual(tracking.history, [
+            .subscription(Subscriptions.empty),
+            .completion(.failure(OtherError(expectedError)))
+        ])
+    }
+
+    func testRange() {
+        // Given
+        let publisher = PassthroughSubject<Int, TestingError>()
+        let mapError = publisher.mapError { OtherError($0) }
+        let tracking = TrackingSubscriberBase<Int, OtherError>(
+            receiveSubscription: { $0.request(.unlimited) }
+        )
+        // When
+        publisher.send(1)
+        mapError.subscribe(tracking)
+        publisher.send(2)
+        publisher.send(3)
+        publisher.send(completion: .finished)
+        publisher.send(5)
+        // Then
+        XCTAssertEqual(tracking.history, [
+            .subscription(Subscriptions.empty),
+            .value(2),
+            .value(3),
+            .completion(.finished)
+        ])
+    }
+
+    func testNoDemand() {
+        // Given
+        let subscription = CustomSubscription()
+        let publisher = CustomPublisher(subscription: subscription)
+        let mapError = publisher.mapError { OtherError($0) }
```

Comment:
Can we use functional syntax instead of a trailing closure here and below?
```suggestion
        let mapError = publisher.mapError(OtherError.init)
```